src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -176,6 +176,8 @@ public void setValidateAbstractClassNames(boolean validateAbstractClassNames) {
             TokenTypes.PARAMETER_DEF,
             TokenTypes.METHOD_DEF,
             TokenTypes.IMPORT,
+            TokenTypes.TYPE_ARGUMENT,
+            TokenTypes.TYPE_UPPER_BOUNDS,
```

Comment:
If we have `TYPE_UPPER_BOUNDS`, then don't we need `TYPE_LOWER_BOUNDS` here too?

Reply:
Fixed, I added `TYPE_LOWER_BOUNDS`